### PR TITLE
add CPDB metadata

### PR DESCRIPTION
**Pull request by damonmcc**
related to https://github.com/NYCPlanning/data-engineering/issues/800

this adds two CPDB metadata files: 
- `commitments` relates to one dataset and one destination
- `projects` relates to three datasets and three destinations

links to those four destinations are in the [issue description](https://github.com/NYCPlanning/data-engineering/issues/800)

tested by running locally with changes from PR https://github.com/NYCPlanning/data-engineering/pull/890

## screenshots showing `Last Updated` on each page
Screenshot 2024-06-18 at 3 57 05 PM

Screenshot 2024-06-18 at 3 57 20 PM

Screenshot 2024-06-18 at 3 57 32 PM

Screenshot 2024-06-18 at 3 57 41 PM

### CODE REVIEW

File: products/cpdb/cpdb/commitments/metadata.yml
Comment (fvankrieken) [inline products/cpdb/cpdb/commitments/metadata.yml:1]: Do we need the double `cpdb` folders?

File: products/cpdb/cpdb/projects/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+    - name: projects_points
+      type: shapefile
+      filename: cpdb_projects_pts.shp.zip
+      overrides:
+        display_name: Capital Projects Database (CPDB) - Projects (Points)
+        omit_columns: []
+        ignore_validation: []
+        columns:
+          maprojid: {name: maprojid}
```
Comment (fvankrieken) [inline products/cpdb/cpdb/projects/metadata.yml:56]: Should we leave out the ones that don't override anything? Or do we think it's better to include all columns like this for consistency. Could go either way

**Comment by fvankrieken** [comment]
This looks like it was a lot of work. Nice!

File: products/cpdb/cpdb/projects/metadata.yml (hunk 1 above)
Comment (damonmcc) [reply products/cpdb/cpdb/projects/metadata.yml:56]: I don't know why I thought we had to put these (@alexrichey?). I'd rather leave em out

File: products/cpdb/cpdb/commitments/metadata.yml
Comment (damonmcc) [reply products/cpdb/cpdb/commitments/metadata.yml:1]: nope! that's an artifact from when I started with one file. will change

File: products/cpdb/cpdb/projects/metadata.yml (hunk 1 above)
Comment (damonmcc) [reply products/cpdb/cpdb/projects/metadata.yml]: I just tried to publish without em and got 

```bash
│ /Users/damonmccullough/github/data-engineering/dcpy/connectors/socrata/publish.py:50 in          │
│ _socrata_request                                                                                 │
│                                                                                                  │
│    47 │   request_fn = getattr(requests, method.lower())                                         │
│    48 │   resp = request_fn(url, auth=_simple_auth(), **kwargs)                                  │
│    49 │   if not resp.ok:                                                                        │
│ ❱  50 │   │   raise Exception(resp.text)   

...

Exception: {"params":{},"message":"You are not allowed to access that resource","key":"forbidden"}
```

File: products/cpdb/commitments/metadata.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      - commitments
+
+package:
+  dataset_files:
+    - name: commitments
+      filename: cpdb_planned_commitments.csv
+      type: csv
+
+  attachments:
+    - cpdb_data_dictionary.xlsx
```
Comment (alexrichey) [inline products/cpdb/commitments/metadata.yml:34]: curious if this pushes correctly. If so, we can strike this [todo](https://github.com/NYCPlanning/data-engineering/blob/7a35e53bd04f0c9eb2bf3f418004f8a8e2aca4dd/dcpy/connectors/socrata/publish.py#L326)

File: products/cpdb/commitments/metadata.yml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    example: Fixed Asset
+  - name: typcname
+    display_name: typcname
+    description: Ten year plan category (typc) the planned commitment supports.
+    data_type: text
+
+  - name: plannedcommit_ccnonexempt
+    display_name: plannedcommit_ccnonexempt
+    description: Sum of City funding (Non-Exempt) associated with the planned commitment in the Capital Commitment Plan. 
+    data_type: integer
```
Comment (alexrichey) [inline products/cpdb/commitments/metadata.yml:113]: Are these actually integers? (or is our validation for integers borked?)

File: products/cpdb/projects/metadata.yml [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+          spent_total: {name: sptotal}
+          geometry: {name: geometry}
+
+  attachments:
+    - cpdb_data_dictionary.xlsx
+    - source_data_versions.csv
+
+columns:
+  - name: ccpversion
+    display_name: ccpversion
```
Comment (alexrichey) [inline products/cpdb/projects/metadata.yml:172]: So with the `display_name`s, do these match what was on Socrata before we pushed? IIRC, the display_name is the column header that Socrata spits out for exported files, e.g. csv's

File: products/cpdb/commitments/metadata.yml (hunk 3 above)
Comment (damonmcc) [reply products/cpdb/commitments/metadata.yml:113]: yup they're integers

Screenshot 2024-06-20 at 11 27 42 AM

**Review by alexrichey — APPROVED** [review]
Good Stuff! Thinking we can either 
- try and figure out the xlsx business now
- make the socrata connector throw an error (for the time being) if an xlsx is added
- comment out the attachment
Either way, this seems gtg

**Comment by damonmcc** [comment]
I'll comment out the attachment. seems nice to have that info even if it isn't used yet

File: products/cpdb/commitments/metadata.yml (hunk 2 above)
Comment (damonmcc) [reply products/cpdb/commitments/metadata.yml]: (discussed, see PR comments below)

**Pull request merged**